### PR TITLE
[Doc] Update Resource Group behavior changes

### PR DESCRIPTION
**Pull request by EsoragotoSpirit**
Signed-off-by: EsoragotoSpirit <wanglichen@starrocks.com>

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: docs/administration/monitor_manage_big_queries.md [hunk 1]
```diff
@@ -20,10 +20,10 @@ StarRocks provides two precautionary instruments for dealing with big queries -
 
 Resource groups can automatically identify and terminate big queries. When creating a resource group, you can specify the upper limit of CPU time, memory usage, or scan row count that a query is entitled to. Among all queries that hit the resource group, any queries that require more resources are rejected and returned with an error. For more information, see [Resource Isolation](../administration/resource_group.md).
 
-Before creating resource groups, you must execute the following statement to enable the resource group feature:
+Before creating resource groups, you must execute the following statement to enable Pipeline Engine, on which the Resource Group feature depends on:
```
Comment (evelynzhaojie) [inline docs/administration/monitor_manage_big_queries.md:23]: two "on", delete `on` at the end of this sentence

File: docs/administration/resource_group.md [hunk 2]
```diff
@@ -148,17 +148,18 @@ You can isolate computing resources among queries by configuring resource groups
...
 ```
 
-If you want to globally enable resource groups, execute the following statement:
-
-```SQL
-SET GLOBAL enable_resource_group = true;
-```
+> **NOTE**
+>
+> From v3.1.0 onwards, Resource Group is enabled by default.
```
Comment (evelynzhaojie) [inline docs/administration/resource_group.md:162]: add this one “Therefore, the enable_resource_group variable is deprecated."

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by evelynzhaojie — APPROVED** [review]

**Review by amber-create — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by EsoragotoSpirit** [comment]
@mergify backport branch-3.1

**Pull request merged**